fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/security/Hash.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.dedis.popstellar.utility.security
+
+import java.nio.charset.StandardCharsets
+import java.security.MessageDigest
+import java.security.NoSuchAlgorithmException
+import java.util.Base64
+import timber.log.Timber
+
+/** SHA256 Hashing Class */
+object Hash {
+  val TAG: String = Hash::class.java.simpleName
+
+  /**
+   * Hash some objects using SHA256. Concatenate the object's string representation following the
+   * protocol's directive. Then hash the obtained string
+   *
+   * @param strs : the strings to hash
+   * @return the hashed data or null if failed to hash
+   * @throws IllegalArgumentException if the data is null or empty
+   * @throws UnsupportedOperationException if SHA-256 MessageDigest is unavailable
+   */
+  @JvmStatic
+  fun hash(vararg strs: String?): String {
```

Comment:
here as well, a bit of vertical spacing wouldn't hurt.

newlines are really helpful in separating logical blocks in a function, improving readability and communicating intention. here there's clearly:
1. a verification of preconditions (the require of `strs.isNotEmpty()`) 
2. a try block, within which:
2.1. initialization of the digest
2.2. looping over string components
2.3. obtain the result and returning
3. a catch block

On a side note, while Kotlin allows to do `return try`, you're not really returning anything from the catch, so I would suggest only using the `return` keyword on line 35 when encoding the hash.